### PR TITLE
[OpaquePointers] Correctly find leader of EC in type scavenger

### PR DESCRIPTION
**Pull request by LU-JOHN**
Correctly find equivalence class in type scavenger.

### CODE REVIEW

**Review by bader — COMMENTED** [review]
@LU-JOHN, please, reduce the test case.
136,110 lines doesn't seem reasonable to validate this change.

**Comment by bader** [comment]
@jcranmer-intel, could you take a look, please?

**Comment by LU-JOHN** [comment]
Testcase reduced with llvm-reduce.

File: test/type-scavenger/equivalence.ll [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+; Incorrect lookup of equivalence class leader caused an assertion failure when
+; processing call instruction to this name
+; CHECK-SPIRV: _ZN4sycl3_V16marrayIiLm5EEixEm
+; CHECK-LLVM: _ZN4sycl3_V16marrayIiLm5EEixEm
+
+target datalayout = "e-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-n8:16:32:64"
+target triple = "spir64-unknown-unknown"
+
+define spir_func void @_ZN4sycl3_V18lgamma_rINS0_6marrayINS0_6detail9half_impl4halfELm5EEENS0_9multi_ptrINS2_IiLm5EEELNS0_6access13address_spaceE0ELNS9_9decoratedE0EEEEENSt9enable_ifIXaaaasr6detailE11is_marray_vIT_Esr6detailE20is_valid_elem_type_vISE_fdS5_Eaaaasr6detailE14is_multi_ptr_vIT0_Esr6detailE26has_writeable_addr_space_vISF_Esr6detailE20is_valid_elem_type_vISF_NS3_15change_elementsIiSE_E4typeEEESE_E4typeESE_SF_() {
```
Comment (jcranmer-intel) [inline test/type-scavenger/equivalence.ll:17]: You can simplify the function names here so they're not overly long Itanium-mangled names.

**Review by jcranmer-intel — APPROVED** [review]
LGTM

File: test/type-scavenger/equivalence.ll (hunk 1 above)
Comment (LU-JOHN) [reply test/type-scavenger/equivalence.ll]: Function names shortened

**Review by MrSidims — APPROVED** [review]

**Pull request merged**